### ISSUE TITLE
compute q and q_aux as FABs in the main MFIter loop

### ISSUE THREAD

**Issue by zingale** [AMReX-Astro/Castro#908]
Instead of having these as MFs, we should compute them on a box-by-box basis as needed in the MFIter loop.  Note: this is essentially what MHD does now.

**Comment by zingale** [comment]
One issue with this is that we currently do `check_for_cfl_violation` in the main CTU driver before we compute the hydro source terms.  This relies on us having the primitive data MF.  If we instead compute the primitive data as needed on a FAB, we'd need to move this check into the main `construct_ctu_hydro_source` MFIter.

Note: MHD currently has this check in the hydro source MFIter.

**Comment by maxpkatz** [comment]
I started looking at this tonight and remembered one reason why we moved it out in the first place: `q` is computed over 4 ghost cells, but after ctoprim, `q` is read-only in the hydro loop. So if we do it inside the hydro loop on a Fab that means we need to compute the ghost zones for every tile, whereas if we do it outside the hydro loop on a MultiFab we only have ghost zones for the box itself. This distinction is not particularly relevant on GPUs because we don't tile there, but it could matter for CPUs. Let's make sure to check performance for the PR that does this before we merge, on both CPUs and GPUs.
